policy-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/policy/kafka/KafkaContractTest.java

```diff
@@ -193,31 +197,42 @@ void testPolicyExistsForValidRequest() {
         results.removeLast();
         assertAll("Results are correct and ordered",
                 () -> assertThat(results)
+                        .as("Check the size of the returned results")
                         .hasSize(1),
 
                 () -> assertThat(results)
                         .allSatisfy(result ->
                                 assertThat(result.headers().lastHeader(Token.HEADER).value())
+                                        .as("Check the bytes of the 'x-request-token' value")
                                         .isEqualTo(ContractTestData.REQUEST_TOKEN.getBytes())),
 
                 () -> assertThat(results.stream()
                         .map(ConsumerRecord::value)
                         .map(response -> response.get("resource").get("type").asInt())
                         .collect(Collectors.toList()))
+                        .as("Check the results are sorted correctly")
                         .isSorted(),
 
-                () -> assertAll("Policy Assertion",
-                        () -> assertThat(results.get(0).value().get("user").get("userId").get("id").asText())
-                                .isEqualTo("test-user-id"),
-                        () -> assertThat(results.get(0).value().get("resourceId").asText())
-                                .isEqualTo("file:/test/resourceId"),
-                        () -> assertThat(results.get(0).value().get("context").get("contents").get("purpose").asText())
-                                .isEqualTo("test-purpose"),
-                        () -> assertThat(results.get(0).value().get("rules").get("message").asText())
-                                .isEqualTo("no rules set"),
-                        () -> assertThat(results.get(0).value().get("rules").get("rules").get("1-PassThroughRule").get("class").asText())
-                                .isEqualTo("uk.gov.gchq.palisade.contract.policy.common.PassThroughRule")
-                ));
+                () -> assertThat(results)
+                        .as("Check the values within the results")
+                        .allSatisfy(result -> {
```

Comment:
allSatisfy is not needed here

Reply:
While there is only 1 result I thought this was a better way to check all the values within the result.